be/src/exec/pipeline/exchange/sink_buffer.h

```diff
@@ -220,33 +276,37 @@ class SinkBuffer {
                 }
             }
         }
-
-        request.params->set_sequence(_request_seq++);
-
+        request.params->set_sequence(_request_seqs[request.fragment_instance_id]++);
         auto* closure = _closures[request.fragment_instance_id];
+        DCHECK(closure != nullptr);
         DCHECK(!closure->has_in_flight_rpc());
         closure->ref();
         closure->cntl.Reset();
         closure->cntl.set_timeout_ms(500);
         closure->cntl.request_attachment().append(request.attachment);
-        _in_flight_rpc_num++;
+        _in_flight_rpc_num.fetch_add(1, std::memory_order_release);
         request.brpc_stub->transmit_chunk(&closure->cntl, request.params.get(), &closure->result, closure);
     }
 
-    // To avoid lock
+    PriorityThreadPool* _io_threads = nullptr;
     MemTracker* _mem_tracker = nullptr;
+
     std::unordered_map<TUniqueId, size_t> _num_sinkers_per_dest_instance;
-    int64_t _request_seq = 0;
+    std::unordered_map<TUniqueId, int64_t> _request_seqs;
     std::atomic<int32_t> _in_flight_rpc_num = 0;
-    std::atomic_bool _is_cancelled = false;
 
     std::unordered_map<TUniqueId, CallBackClosure<PTransmitChunkResult>*> _closures;
     std::unordered_map<TUniqueId, std::queue<TransmitChunkInfo>> _buffers;
-    std::condition_variable _buffer_empty_cv;
-    std::mutex _mutex;
 
-    std::thread _thread;
-    std::atomic_bool _is_finished = false;
+    std::mutex _mutex;
+    bool _is_finishing = false;
+    bool _need_one_more_scan = false;
+    bool _is_io_task_active = false;
```

Comment:
Rename to `_is_sink_task_active` or `_is_rpc_task_active` ?

Reply:
ok